### PR TITLE
Migrate deprecated buildDir

### PR DESCRIPTION
**Pull request by Goooler**

### CODE REVIEW

File: build-logic/src/main/groovy/org.apache.groovy-asciidoctor.gradle [hunk 1]
```diff
@@ -53,14 +53,14 @@ asciidoctor {
...
-    outputDir = "$buildDir/asciidoc/html5"
+    outputDir = layout.buildDirectory.dir("asciidoc/html5")
     resources {
         duplicatesStrategy = DuplicatesStrategy.WARN
 
         from("${rootProject.projectDir}/src/spec/doc/assets") {
             include 'css/style.css'
         }
-        from("$buildDir/tmp/asciidoctor-assets")
+        from layout.buildDirectory.dir("asciidoc/html5/assets")
```
Comment (paulk-asert) [inline build-logic/src/main/groovy/org.apache.groovy-asciidoctor.gradle:63]: Should this be the same as the next change, i.e. with tmp in the path?

Comment (Goooler) [reply build-logic/src/main/groovy/org.apache.groovy-asciidoctor.gradle:63]: Ooops, typo here

Comment (Goooler) [inline build-logic/src/main/groovy/org.apache.groovy-asciidoctor.gradle:63]:
```suggestion
        from layout.buildDirectory.dir("tmp/asciidoctor-assets")
```

File: build-logic/src/main/groovy/org.apache.groovy-asciidoctor.gradle [hunk 2]
```diff
@@ -126,7 +126,7 @@ asciidoctor.onlyIf { project.file('src/spec/doc').exists() }
 
 def asciidoctorAssets = tasks.register("asciidoctorAssets", Copy) {
     from project.fileTree('src/spec/doc/assets')
-    into "$buildDir/tmp/asciidoctor-assets"
+    into layout.buildDirectory.dir("tmp/asciidoc/html5/assets")
```
Comment (Goooler) [inline build-logic/src/main/groovy/org.apache.groovy-asciidoctor.gradle:129]:
```suggestion
    into layout.buildDirectory.dir("tmp/asciidoctor-assets")
```

File: build.gradle [hunk 3]
```diff
@@ -254,7 +254,7 @@ sonarqube {
         property "sonar.organization", "apache"
         property "sonar.host.url", "https://sonarcloud.io"
         property "sonar.cpd.exclusions", "**/*"
-//        property "sonar.coverage.jacoco.xmlReportPaths", "${rootProject.buildDir}/reports/jacoco/jacocoAllReport/jacocoAllReport.xml"
+//        property "sonar.coverage.jacoco.xmlReportPaths", layout.buildDirectory.dir("reports/jacoco/jacocoAllReport/jacocoAllReport.xml").get().asFile.absolutePath
```
Comment (Goooler) [inline build.gradle:257]:
```suggestion
//        property "sonar.coverage.jacoco.xmlReportPaths", layout.buildDirectory.file("reports/jacoco/jacocoAllReport/jacocoAllReport.xml").get().asFile.absolutePath
```

File: build-logic/src/main/groovy/org.apache.groovy-distribution.gradle [hunk 4]
```diff
@@ -181,7 +181,7 @@ tasks.register('dist') {
 
 tasks.register("installGroovy", Sync) {
     duplicatesStrategy = DuplicatesStrategy.FAIL
-    def installDir = sharedConfiguration.installationDirectory.orElse(project.layout.buildDir.file("install"))
+    def installDir = sharedConfiguration.installationDirectory.orElse(layout.buildDirectory.file("install").get().asFile)
```
Comment (paulk-asert) [inline build-logic/src/main/groovy/org.apache.groovy-distribution.gradle:184]: buildDirectory.file should probably be buildDirectory.dir here as well.

File: build-logic/src/main/groovy/org.apache.groovy-performance.gradle [hunk 5]
```diff
@@ -70,7 +70,7 @@ tasks.named('compileJmhGroovy') {
 
 tasks.register('displayJmhResults') {
     doLast {
-        println file("$buildDir/results/jmh/results.txt").text
+        println layout.buildDirectory.file("results/jmh/results.txt").get().asFile
```
Comment (paulk-asert) [inline build-logic/src/main/groovy/org.apache.groovy-performance.gradle:73]: This currently actually prints out the file, not the name as proposed, so the .text is still needed.

Comment (Goooler) [reply build-logic/src/main/groovy/org.apache.groovy-performance.gradle:73]: Good catch!

**Pull request merged**